cmake/MakeChm.cmake

```diff
@@ -1,69 +0,0 @@
-# Create CHM help files for Win32
-# Procedure translated to cmake from make_chm() in gnucash-on-windows.git:install-impl.sh,
-# originally written by Andreas Köhler.
-
-
-set(chmfile "${docname}.chm")
-set(mapfile "${docname}.hhmap")
-set(htmlhelpdir "${CURRENT_BIN_DIR}/htmlhelp")
-
-#string(REPLACE ";" " " xlstprocflagslst "${XSLTPROCFLAGS}")
-#message(STATUS "XSLTPROCFLAGS: ${XSLTPROCFLAGS}\nxlstprocflagslst: ${xlstprocflagslst}")
-
-set(htmlhelp_xsl "http://docbook.sourceforge.net/release/xsl/current/htmlhelp/htmlhelp.xsl")
-
-execute_process(
-    # FIXME Reusing ${XSLTPROCFLAGS} fails as xsltproc gets this as one single parameter instead of 3...
-    COMMAND ${XSLTPROC} --path "${SRC_DIR}/docbook" --xinclude
-                        --stringparam htmlhelp.chm ${chmfile}
-                        "${SRC_DIR}/xsl/1.79.2/htmlhelp/htmlhelp.xsl"
-                        "${CURRENT_SRC_DIR}/${docname}.xml"
-    WORKING_DIRECTORY "${htmlhelpdir}")
-
-file(COPY "${CURRENT_SRC_DIR}/figures" DESTINATION "${htmlhelpdir}")
-
-set(count 0)
-set(HPP "")
-list(APPEND HHP "" "[ALIAS]" "IDH_0=index.html")
-set(MAP "")
-list(APPEND MAP "" "[MAP]" "#define IDH_0 0")
-set(HHMAP "[MAP]")
-message(STATUS "Searching for anchors...")
-
-file(GLOB allhtml RELATIVE "${htmlhelpdir}" "${htmlhelpdir}/*.html")
-foreach(src_xml ${entities})
-    file(STRINGS "${CURRENT_SRC_DIR}/${src_xml}" sectlines REGEX "sect.*id=[\"'][^\"']*[\"']")
-    foreach(sectline ${sectlines})
-        string(REGEX REPLACE ".*sect.*id=[\"']([^\"']*)[\"'].*" "\\1" sectid ${sectline})
-        #message(STATUS "allhtml: ${allhtml}\nsectline: ${sectline}\nsrc_xml: ${sectid}")
-        unset(html_id_file)
-        foreach(htmlfile ${allhtml})
-            file(STRINGS "${htmlhelpdir}/${htmlfile}" searchresult REGEX "[\"']${sectid}[\"']")
-            if(searchresult)
-                set(html_id_file ${htmlfile})
-                break()
-            endif()
-        endforeach()
-        if(html_id_file)
-            MATH(EXPR count "${count}+1")
-            list(APPEND HHP "IDH_${count}=${html_id_file}#${sectid}")
-            list(APPEND MAP "#define IDH_${count} ${count}")
-            list(APPEND HHMAP "${sectid}=${count}")
-        endif()
-    endforeach()
-endforeach()
-
-set(HHP "${HHP};${MAP}")
-string(REPLACE ";" "\n" HHP_OUT "${HHP}")
-string(REPLACE ";" "\n" HHMAP_OUT "${HHMAP}")
-file(APPEND "${htmlhelpdir}/htmlhelp.hhp" ${HHP_OUT})
-file(WRITE "${BUILD_DIR}/${mapfile}" ${HHMAP_OUT})
```

Comment:
I am aware  this PR is closed and you will resubmit this work in a future PR. But just for the record I am not convinced thhe code to generate the hhp and hhmap files  can just be  discarded. GnuCash parses the hhmap file in gnucash/gnome-utils/gnc-help-utils.c. So this change should be tested a bit more before it can be applied. Be sure not to mix this in the same commit t hat will drop the xsl stylesheet from gnucash-docs so they can be reviewed independently.

Reply:
> GnuCash doesn't `#include <htmlhelp.h>` when it is build.

It does include this, in the same gnc-help-utils.c file.

Reply:
It is my oversight.

Then, where is "context.h" and "alias.h"? I can't find them. They are used for context-sensitive help. For example, when we press the "HELP" button on the Scheduled Transaction window, if it exists, the Scheduled Transaction page of GnuCash-help will open. And it is not necessary to create [map] and [alias] sections by our hands.

Just copy them as "context.h" and "alias.h", and run `make chm`, [map] and [alias] section will be created automatically by xsltproc.